syntax = "proto3";

package queue;

service QueueService {
  // Start the queue with a given name
  rpc StartQueue(StartQueueRequest) returns (StartQueueResponse);

  // Consume the queue with a given name (streaming)
  rpc ConsumeQueue(ConsumeQueueRequest) returns (stream ConsumeQueueResponse);

  // Add a message to the queue
  rpc AddMessage(AddMessageRequest) returns (AddMessageResponse);

}

// Request and response messages for starting the queue
message StartQueueRequest {
  string queue_name = 1;
}

message StartQueueResponse {
  string status = 1;
}

// Request and response messages for consuming the queue
message ConsumeQueueRequest {
  string queue_name = 1;
}

message ConsumeQueueResponse {
  bytes message = 1; // Single message at a time
}

// Request and response messages for adding a message
message AddMessageRequest {
  string queue_name = 1;
  bytes message = 2;
}

message AddMessageResponse {
  string status = 1;
}


